src/propertySelector/propertySelector.js

```diff
@@ -0,0 +1,150 @@
+                maxHeight = $container.parent().height() - top - parentGap;
+            }
+            if (typeof top === 'undefined') {
+                maxHeight = $container.parent().height() - bottom - parentGap;
+            }
+            $container.css({ top, left, right, bottom, maxHeight });
+        },
+
+        /**
+         * Updates the list
+         */
+        redrawList: function redrawList() {
+            $propertyListContaner.empty();
+            availableProperties.forEach(property => {
+                property.selected = selectedProperties.has(property.id);
+                if (search === '' || property.label.toLowerCase().includes(search.toLowerCase())) {
+                    $propertyListContaner.append(createPropertyOption(property));
+                }
+            });
+        },
+        /**
+         * Adds and setups buttons to button container
+         */
+        addButtons: function addButtons() {
+            const cancelButton = buttonFactory({
+                id: 'cancel',
+                label: 'Cancel',
+                type: 'info',
+                cls: 'btn-secondary'
+            }).on('click', () => {
+                this.trigger('cancel');
+            });
+
+            const saveButton = buttonFactory({
+                id: 'save',
+                label: 'Save',
+                type: 'info'
+            }).on('click', () => {
+                this.trigger('select', [...selectedProperties]);
+            });
+
+            cancelButton.render($buttonsContainer);
+            saveButton.render($buttonsContainer);
+        },
+        /**
+         * Setups search input event listners
+         */
+        setupSearch: function setupSearch() {
+            $searchInput = $('input.search-property', $container);
+            $searchInput.on('input', function () {
+                search = $(this).val();
+                instance.redrawList();
+            });
+        }
+    })
+        .setTemplate(propertySelectorTpl)
+        .on('render', function () {
+            //component parts reference assignments
+            $container = instance.getElement();
+            $propertyListContaner = $('.property-list-container', $container);
+            $buttonsContainer = $('.control-buttons-container', $container);
+
+            this.positionContainer();
+
+            this.redrawList();
+
+            this.setupSearch();
+
+            this.addButtons();
+
+            this.trigger('ready');
+        })
+        .on('init', function () {
+            //setup data
+            const data = instance.config.data;
+            if (data.available) {
+                if (!Array.isArray(data.available)) {
+                    availableProperties = Object.values(data.available);
+                } else {
+                    availableProperties = data.available;
+                }
+            }
+            if (data.selected) {
+                selectedProperties = new Set(data.selected);
+            }
```

Comment:
**issue:** As I see, the `selectedProperties` may not be initialized. In this context, the rendering part of the component may fail since it relates to this variable. We may have available properties but no selection, but the `redrawList` expects some.

Reply:
Thanks, now I'm initialising the set unconditionally